arctic/store/version_store.py

```diff
@@ -79,6 +79,8 @@ def _ensure_index(self):
                                          background=True)
         collection.versions.create_index([('symbol', pymongo.ASCENDING), ('version', pymongo.DESCENDING)], unique=True,
                                          background=True)
+        collection.versions.create_index([('symbol', pymongo.ASCENDING), ('version', pymongo.DESCENDING), ('metadata.deleted', pymongo.ASCENDING)],
```

Comment:
As this compound index extends the previous one, I think you can remove the one on line 80-81 as this will be used instead (and means you don't need to maintain 2 indexes).

Reply:
The index on 80-81 is a unique index. We can't make this one unique as it will allow both a deleted and active version to exist. And we can't remove the index on 80-81 for a similar reason of allowing 2 of the same version to exist.